src/openms/source/TRANSFORMATIONS/FEATUREFINDER/FeatureFinderMultiplexAlgorithm.cpp

```diff
@@ -755,54 +755,53 @@ namespace OpenMS
     std::vector<MultiplexDeltaMasses> masses = generator.getDeltaMassesList();
     std::vector<MultiplexIsotopicPeakPattern> patterns = generatePeakPatterns_(charge_min_, charge_max_, isotopes_per_peptide_max_, masses);
     
-    std::vector<MultiplexFilteredMSExperiment> filter_results;
     if (centroided)
     {
       // centroided data
+      
+      /**
+       * filter for peak patterns
+       */
       MultiplexFilteringCentroided filtering(exp_centroid_, patterns, isotopes_per_peptide_min_, isotopes_per_peptide_max_, param_.getValue("algorithm:intensity_cutoff"), param_.getValue("algorithm:rt_band"), param_.getValue("algorithm:mz_tolerance"), (param_.getValue("algorithm:mz_unit") == "ppm"), param_.getValue("algorithm:peptide_similarity"), param_.getValue("algorithm:averagine_similarity"), param_.getValue("algorithm:averagine_similarity_scaling"), param_.getValue("algorithm:averagine_type"));
       filtering.setLogType(getLogType());
-      filter_results = filtering.filter();
-    }
-    else
-    {
-      // profile data
-      MultiplexFilteringProfile filtering(exp_profile_, exp_centroid_, boundaries_exp_s, patterns, isotopes_per_peptide_min_, isotopes_per_peptide_max_, param_.getValue("algorithm:intensity_cutoff"), param_.getValue("algorithm:rt_band"), param_.getValue("algorithm:mz_tolerance"), (param_.getValue("algorithm:mz_unit") == "ppm"), param_.getValue("algorithm:peptide_similarity"), param_.getValue("algorithm:averagine_similarity"), param_.getValue("algorithm:averagine_similarity_scaling"), param_.getValue("algorithm:averagine_type"));
-      filtering.setLogType(getLogType());
-      filter_results = filtering.filter();
-    }
-
-    /**
-     * cluster filter results
-     */
-    std::vector<std::map<int, GridBasedCluster> > cluster_results;
-    if (centroided)
-    {
-      // centroided data
+      std::vector<MultiplexFilteredMSExperiment> filter_results = filtering.filter();
+      
+      /**
+       * cluster filter results
+       */
       MultiplexClustering clustering(exp_centroid_, param_.getValue("algorithm:mz_tolerance"), (param_.getValue("algorithm:mz_unit") == "ppm"), param_.getValue("algorithm:rt_typical"), static_cast<double>(param_.getValue("algorithm:rt_min")));
       clustering.setLogType(getLogType());
-      cluster_results = clustering.cluster(filter_results);
+      std::vector<std::map<int, GridBasedCluster> > cluster_results = clustering.cluster(filter_results);
+
+      /**
+       * construct feature and consensus maps i.e. the final results
+       */
+      filtering.getCentroidedExperiment().swap(exp_centroid_);
```

Comment:
what should this line do? swapping exp_centroid_ with the copy?

Reply:
is this what fixes the bug?

Reply:
`exp_centroid_` is updated by the (intensity filtered) centroid data from the `filtering` object.

Reply:
@hroest Yes, the filter results are stored as [indices](https://github.com/OpenMS/OpenMS/blob/develop/src/openms/include/OpenMS/TRANSFORMATIONS/FEATUREFINDER/MultiplexSatelliteCentroided.h#L61) in the **filtered** centroid data. Hence, I update this member variable before constructing the maps.